src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx

```diff
@@ -163,13 +163,18 @@ const ReleaseDataUploadsSection = ({ publicationId, releaseId }: Props) => {
 
   const handleSubmit = useFormSubmit<FormValues>(async (values, actions) => {
     setIsUploading(true);
-    await editReleaseDataService.uploadDataFiles(releaseId, {
-      subjectTitle: values.subjectTitle,
-      dataFile: values.dataFile as File,
-      metadataFile: values.metadataFile as File,
-    });
-    setIsUploading(false);
-    await resetPage(actions);
+    await editReleaseDataService
+      .uploadDataFiles(releaseId, {
+        subjectTitle: values.subjectTitle,
+        dataFile: values.dataFile as File,
+        metadataFile: values.metadataFile as File,
+      })
+      .then(() => {
+        resetPage(actions);
```

Comment:
Shouldn't have to change this all to use promise syntax. Wrapping the entire thing with a try/finally would have sufficed. Async/await and promises are all interchangeable, but I personally prefer async/await in the majority of cases.

If you do want to continue using promise syntax, then we should at least return the `resetPage` here as it is an async task itself and we wouldn't want to `setIsUploading(false)` until after the page has finished resetting i.e.

```typescript
.then(() => resetPage(actions));
```

Reply:
Have changed the logic slightly to ensure to clear the spinner before resetting the form which is the only way with the current error handling logic that works.

Reply:
Cool, not going to delve too deep into it, but it seems like this new change should also be applied to `RleeaseDataUploadsSection` too?